### DISCUSSION TITLE
After resetting a module, it should be where it was originally after your first install of the theme

### DISCUSSION THREAD

**Discussion by khouloudbelguith** [PrestaShop/PrestaShop#35512] [Ideas]
#### Describe the bug

If you reset the module or if you uninstall the module, clone the module from GitHub and install it, then on FO, the slider is hooked on the wrong display.

Checked with `ps_imageslider` module, `ps_featuredproducts` module, `ps_customtext` module, `Banner` module, `ps_emailsubscription` module, `ps_socialfollow` module, `ps_categorytree` Module, `productcomments` Module, `ps_customersignin` module

#### Steps to Reproduce

1. Go to BO > Module Manager page > Search for the slider module
2. Click on reset
3. Go to FO > Home page
4. See error => the slider is hooked on the wrong display.

Before
![image](https://user-images.githubusercontent.com/16067358/91424150-b7b44700-e859-11ea-9d11-d28d97d6cfa0.png)

After
![image](https://user-images.githubusercontent.com/16067358/91424113-acf9b200-e859-11ea-808c-208c77aabf85.png)

#### Additional information

* PrestaShop version: 1770beta2build1, 1767
* PHP version: 7.2

#### Wanted behavior

After resetting a module, it should be where it was originally after your first install of the theme.
That's the point of resetting something: putting it back to its original state.

[a comment by an automated account is not shown]

**Comment by khouloudbelguith** [comment]
Related to https://github.com/PrestaShop/PrestaShop/issues/16968

**Comment by PierreRambaud** [comment]
I'm not sure it's a bug :thinking: You reset or reinstall a module, the position is automatically set at the end, nothing predicted that when you install/reset these modules your theme want to be set at position X.

You've access to the Module Positions page to change the value.

**Comment by Progi1984** [comment]
@PrestaShop/product-team What do you think about @PierreRambaud answer ?

**Comment by marionf** [comment]
After resetting a module, it should be where it was originally after your first install of the theme.
That's the point of resetting something: putting it back to its original state.

**Comment by ghost** [comment]
Hi @marionf for the #pullrequestmonday i want to work on it but I need to know if it's only for classic theme or for all theme because the work to do it's very different.

I agree @PierreRambaud because : 

The Merchant : 

1 ) he Install a theme , 
2 ) he install a new module ( sales prodcuts ) an he put in at position one.
3 ) After to week he reset ps_imageslider 

The shop dont respect the merchant position ?

**Comment by ghost** [comment]
And it's possible to set the defaut position only if the module is include with the theme.

**Comment by jeckyl** [comment]
> After resetting a module, it should be where it was originally after your first install of the theme.
> That's the point of resetting something: putting it back to its original state.

How to handle this if the user has made the decision to move the module to another location?

**Comment by ghost** [comment]
@jf-viguier say  us an option exist for theme reset  and all positions are good after. But for this option we need to put PS_MODE_DEV  to ON why not allow this option for all context ?

**Comment by jeckyl** [comment]
@jf-viguier picture
![image](https://user-images.githubusercontent.com/595633/116867958-36922880-ac0e-11eb-9870-a0f0781b7a58.png)

**Comment by jf-viguier** [comment]
Yes, a theme reset should solve the issue if theme.yml is correct.

**Comment by matks** [comment]
Hello @jf-viguier @jeckyl @okom3pom what is your idea?

I think your plan is:
- when theme X is enabled
- module B, used in theme X is reset
- using theme.yml from theme X we can find the wanted position of the module

Is that your plan?

If yes it means any module not included in theme.yml will continue using current behavior: after reset it is hooked at the end.

If @PrestaShop/product-team agree with this plan, I suggest to add a small confirmation window before doing it. "Hello, you are resetting the module, do you prefer module is hooked 1) at the end 2) at position X as configured in your theme"

Asking the user what he would prefer is a good way to avoid receiving a bug report "Hello, PrestaShop resets module A to its theme position, this is a bug, I don't want that..." 😆

**Comment by ghost** [comment]
Give the choice for the user is the best way!

**Comment by Mch0** [comment]
im agree with you @matks, use theme yml if entry exist with the module else, show confirmation windows for the user to choice the position for the module

**Comment by Hlavtox** [comment]
@kpodemski I think we can close this, that is what just happens and if you go that far to reset a module, you can move it back..

**Comment by MatShir** [comment]
More a feature request, should we it into discussion then ? @Hlavtox
